src/server/game/Spells/SpellEffects.cpp

```diff
@@ -2977,7 +2977,7 @@ void Spell::EffectSummonPet(SpellEffIndex effIndex)
 
     float x, y, z;
     owner->GetClosePoint(x, y, z, owner->GetCombatReach());
-    Pet* pet = owner->SummonPet(petentry, x, y, z, owner->GetOrientation(), SUMMON_PET, 0);
+    Pet* pet = owner->SummonPet(petentry, x, y, z, owner->GetOrientation(), SUMMON_PET, 0, true);
```

Comment:
https://github.com/TrinityCore/TrinityCore/pull/23271/files#diff-b3ff7e4adf5326d9790d99d1dee0da2dR2945 should return an error message about pet being dead too